server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java

```diff
@@ -1467,26 +1473,30 @@ private void deleteDataInFiles(Collection<TsFileResource> tsFileResourceList, De
       // delete data in memory of unsealed file
       if (!tsFileResource.isClosed()) {
         TsFileProcessor tsfileProcessor = tsFileResource.getUnsealedFileProcessor();
-        tsfileProcessor.deleteDataInMemory(deletion);
+        tsfileProcessor.deleteDataInMemory(deletion, devicePaths);
       }
 
       // add a record in case of rollback
       updatedModFiles.add(tsFileResource.getModFile());
     }
   }
 
-  private void tryToDeleteLastCache(PartialPath deviceId, String measurementId, long startTime,
-      long endTime) throws WriteProcessException {
+  private void tryToDeleteLastCache(PartialPath deviceId, PartialPath originalPath,
+      long startTime, long endTime) throws WriteProcessException {
+    if (!IoTDBDescriptor.getInstance().getConfig().isLastCacheEnabled()) {
+      return;
+    }
     try {
       MManager manager = MManager.getInstance();
```

Comment:
help me change the MManager.getInstance() to IoTDB.metaManager

Reply:
Fixed